SilKit/source/util/StringHelpers.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2022 Vector Informatik GmbH
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+#include "StringHelpers.hpp"
+
+
+namespace SilKit {
+namespace Util {
+
+
+auto EscapeString(const std::string& input) -> std::string
+{
+    std::string result;
+    result.reserve(input.size() * 2); // Reserve enough memory for the result
+
+    size_t i = 0;
+    while (i < input.size())
+    {
+        char character = input[i];
+        switch (character)
+        {
+            case '\\':
```

Comment:
During testint I noticed that the escaping code is missing the control characters directly supported by JSON.

```suggestion
            case '\b':
            {
                result += "\\";
                result += 'b';
                break;
            }
            case '\t':
            {
                result += "\\";
                result += 't';
                break;
            }
            case '\n':
            {
                result += "\\";
                result += 'n';
                break;
            }
            case '\f':
            {
                result += "\\";
                result += 'f';
                break;
            }
            case '\r':
            {
                result += "\\";
                result += 'r';
                break;
            }
            case '\\':
```

Reply:
The [JSON spec](https://www.json.org/json-en.html) for reference.